bitnami/thanos/README.md

```diff
@@ -120,6 +120,19 @@ Check the section [Integrate Thanos with Prometheus and Alertmanager](#integrate
 | `existingObjstoreSecret`                      | Secret with Objstore Configuration                                                                                  | `""`                |
 | `existingObjstoreSecretItems`                 | Optional item list for specifying a custom Secret key. If so, path should be objstore.yml                           | `[]`                |
+| `httpConfig`                                  | The [https and basic auth configuration](https://thanos.io/tip/operating/https.md/)                                 | `""`                |
+| `existingHttpConfigSecret`                    | Secret containing the HTTPS and Basic auth configuration                                                            | `""`                |
+| `https.enabled`                               | Set to true to enable HTTPS. Requires a secret containing the certificate and key.                                  | `false`             |
+| `https.autoGenerated`                         | Create self-signed TLS certificates.                                                                                | `false`             |
+| `https.existingSecret`                        | Existing secret containing your own server key and certificate                                                      | `""`                |
+| `https.certFilename`                          |                                                                                                                     | `tls.crt`           |
+| `https.keyFilename`                           |                                                                                                                     | `tls.key`           |
+| `https.caFilename`                            |                                                                                                                     | `ca.crt`            |
+| `https.key`                                   | TLS Key for Thanos HTTPS - ignored if existingSecret is provided                                                    | `""`                |
+| `https.cert`                                  | TLS Certificate for Thanos HTTPS - ignored if existingSecret is provided                                            | `""`                |
+| `https.ca`                                    | (Optional, used for client) CA Certificate for Thanos HTTPS - ignored if existingSecret is provided                 | `""`                |
+| `https.clientAuthType`                        | Server policy for client authentication using certificates. Maps to ClientAuth Policies.                            | `""`                |
+| `auth.basicAuthUsers`                         | Object containing <user>:<passwords> key-value pairs for each user that will have access via basic authentication   | `{}`                |
```

Comment:
As this is containing passwords, I think we should use a secret, or at least, provide the option to use a secret instead.

Reply:
The configuration file is stored in a Secret instead of a configmap because of that reason.

User can provide its configuration file using `existingHttpConfigSecret`.

The main issue is that, if the basic authentication is enabled, the password is also needed for the Probes to succeed, and Thanos uses a scratch container, so no logic can be added to handle the password using env variables.

Reply:
Please refer to the README.md section below, where I tried to depict this same situation.